go/vt/vtgate/planbuilder/ordered_aggregate.go

```diff
@@ -303,6 +303,31 @@ func (oa *orderedAggregate) needDistinctHandling(pb *primitiveBuilder, funcExpr
 	return true, innerAliased, nil
 }
 
+// needDistinctHandling returns true if oa needs to handle the distinct clause.
+// If true, it will also return the aliased expression that needs to be pushed
+// down into the underlying route.
+func (oa *orderedAggregate) needDistinctHandlingGen4(funcExpr *sqlparser.FuncExpr, opcode engine.AggregateOpcode, semTable *semantics.SemTable, vschema ContextVSchema) (bool, *sqlparser.AliasedExpr, error) {
```

Comment:
Comment should be `// needDistinctHandlingGen4  ...`

Reply:
we have very little gen4 code in these files. I think this method should live on `horizon_planning.go` and not here